src/Type/Accessory/HasOffsetValueType.php

```diff
@@ -165,6 +165,15 @@ public function getValuesArray(): Type
 		return new NonEmptyArrayType();
 	}
 
+	public function flipArray(): Type
+	{
+		if ($this->valueType instanceof ConstantIntegerType || $this->valueType instanceof ConstantStringType) {
```

Comment:
Do `ArrayType::castToArrayKeyType()` first :)

Reply:
adapted in a force-push. I also found another special case where the same value is existing twice that does not behave correct at them moment. I'm not sure which code's responsibility is it to figure that out or if it's a big problem, but I documented it in the test

Reply:
maybe that's a bug in `TypeCombinator::intersect()` with `HasOffsetValueType`

Reply:
I think this could be implemented by doing `$type->uniqueArray($type->reverseArray())->flipArray()` :D see https://3v4l.org/EbUU1

but both unique and reverse are not existing yet, so I could look into this later maybe, or we keep it open for now and I adapt it when the time comes. it's a bit special for sure though

Reply:
reverse is the wrong thing then, this will mess with other types. so it'd need to done differently. since `ConstantArray::flipArray()` is implemented correctly I'm really not sure how to do this. maybe it's not feasible

Reply:
A bug like that is fine for me now :)